components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -2814,6 +2815,37 @@ public Set<AccessTokenDO> getActiveTokenSetWithTokenIdByConsumerKeyForOpenidScop
         return activeAccessTokenDOSet;
     }
 
+    /**
+     * Retrieves active AccessTokenDOs with token id for the given consumer key.
+     *
+     * @param consumerKey client id
+     * @return access token data object set
+     * @throws IdentityOAuth2Exception
+     */
+    @Override
+    public Set<AccessTokenDO> getActiveTokenSetWithTokenIdByConsumerKeyAndScope(String consumerKey, List<String> scopes)
+            throws IdentityOAuth2Exception {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Retrieving active access token set with token id of client: " + consumerKey);
+        }
+        Set<AccessTokenDO> activeAccessTokenDOSet = null;
+        for (String scope: scopes) {
+            activeAccessTokenDOSet = getActiveAccessTokenSetByConsumerKeyForScope(consumerKey,
+                    IdentityUtil.getPrimaryDomainName(), scope);
```

Comment:
Any reason why we get active tokens only for the primary user domain?

Reply:
Generally we store access token in Identity database in primary domain, but there are case where access tokens can be partitioned in to secondary user stores. To enable access token partitioning we need enable it in toml config
```
EnableAccessTokenPartitioning
```
In line number 2837 the code above behaviour and call all available user store domain and get access token from them.